frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java

```diff
@@ -92,9 +94,67 @@ public int getNumRunningWorkers(ModelVersionName modelVersionName) {
             CompletableFuture<Integer> future = new CompletableFuture<>();
             int minWorker = model.getMinWorkers();
             int maxWorker = model.getMaxWorkers();
+            WorkerManagerThread workerManagerThread;
             List<WorkerThread> threads;
             if (minWorker == 0) {
+
+                logger.info("Min Worker is 0 - DHANAK");
+                workerManagerThread = workerManagers.remove(model.getModelVersionName());
                 threads = workers.remove(model.getModelVersionName());
+
+                if (workerManagerThread != null) {
+
+                    logger.info("Min Worker is not null - DHANAK");
+
+                    for (WorkerThread thread : threads) {
+                        workerManagerThread.scaleDown(thread.getLifeCycle().getPort());
+                        try {
+                            String cmd =
+                                    String.format(
+                                            OSUtils.getKillCmd(thread.getLifeCycle().getPid()));
+                            Process workerkillprocess = Runtime.getRuntime().exec(cmd, null, null);
+                            workerkillprocess.waitFor(
+                                    configManager.getUnregisterModelTimeout(), TimeUnit.SECONDS);
+                            logger.info("Worker Terminated - DHANAK " + cmd);
+                        } catch (InterruptedException | IOException e) {
+                            logger.warn(
+                                    "WorkerManagerThread interrupted during waitFor, possible async resource cleanup.");
+                            future.complete(HttpURLConnection.HTTP_INTERNAL_ERROR);
+                            return future;
+                        }
+                    }
```

Comment:
The above steps are 
1. kill backend worker process.
2. kill frontend worker thread.
what if the frontend worker thread still sends msg to the backend died worker?

Reply:
The moment the backend worker is killed, The frontend thread would not be able to establish a socket connection to the backend.